parsers/ESIOS.py

```diff
@@ -55,7 +61,10 @@ def fetch_exchange(
     }
 
     zone_key = ZoneKey("->".join(sorted([zone_key1, zone_key2])))
-    if zone_key not in EXCHANGE_ID_MAP.keys():
+    if (
+        zone_key not in EXCHANGE_ID_MAP.keys()
+        or zone_key not in EXCHANGE_GRANULARITY_FACTOR_MAP.keys()
```

Comment:
```suggestion
        or zone_key not in EXCHANGE_GRANULARITY_FACTOR_MAP
```
No need for the `keys` method, you can directly make sure it's here.